### PR TITLE
[master] Initial work for the new logging setup for salt

### PR DESCRIPTION
**Pull request by s0undt3ch**
### What does this PR do?
Starts moving code around to support the new Salt logging support.
This is part of https://github.com/saltstack/salt/pull/55043 so that it gets implemented/reviewed in smaller chunks.

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/pull/55043

### Commits signed with GPG?

Yes/No

### CODE REVIEW

**Review by Akm0d — APPROVED** [review]

File: salt/_logging/__init__.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+'''
+    salt._logging
+    ~~~~~~~~~~~~~
+
+    This is salt's new logging setup.
+    As the name suggests, this is considered an internal API which can change without notice,
+    although, as best effort, we'll try not to break code relyin on it.
```
Comment (waynew) [inline salt/_logging/__init__.py]:
```suggestion
    although, as best effort, we'll try not to break code relying on it.
```

File: salt/_logging/__init__.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+    The ``salt._logging`` package should be imported as soon as possible since salt teaks
```
Comment (waynew) [inline salt/_logging/__init__.py]:
```suggestion
    The ``salt._logging`` package should be imported as soon as possible since salt tweaks
```

**Review by waynew — CHANGES_REQUESTED** [review]
I see that your PR is in chunks now 😂

File: salt/_logging/__init__.py (hunk 1 above)
Comment (s0undt3ch) [reply salt/_logging/__init__.py]: Will fix

Comment (s0undt3ch) [reply salt/_logging/__init__.py]: Fixed

File: salt/_logging/__init__.py (hunk 2 above)
Comment (s0undt3ch) [reply salt/_logging/__init__.py]: Fixed

**Review by waynew — APPROVED** [review]
😞emoji, y u no have :log:???

**Review by dwoz — APPROVED** [review]

**Pull request merged**